File: Colas/Arreglos/Cpp/cola.cpp
#include <iostream>
using namespace std;

#define MAXSIZE 5

int queue[MAXSIZE];
int front = -1, rear = -1;
void insertar(){
    int elemento;
    cout << "\nIngrese el elemento: ";
    cin >> elemento;
    if(rear == MAXSIZE -1){
        cout << "\nOVERFLOW";
        return;
    }
    if(front == -1  && rear == -1){
        front = rear = 0;
    }else{
        rear++;
    }
    queue[rear] = elemento;
    cout << "\nElemento insertado correctamente.\n";
}
void eliminar(){
    if(front == -1 || front > rear){
        cout << "\nUnderflow";
        return;
    }
    int elemento = queue[front];
    if(front == rear){
        front = rear = -1;
    }else{
        front++;
    }
    cout << "\nElemento eliminado: " << elemento << endl;
}
void mostrar(){
    if(rear == -1 || front == -1 || front > rear){
        cout << "\nLa cola esta vacia.\n";
    }else{
        cout << "\nElementos en la cola:\n";
        for (int i = front; i <= rear; i++){
            cout << queue[i] << endl;
        }
    }
}
int main(){
    int opcion = 0;
    while (opcion != 4){
        cout << "\n==========================MENU PRINCIPAL==========================\n";
        cout << "====================================================================\n";
        cout << "1. Insertar\n2. Eliminar\n3. Mostrar\n4. Salir\nIngrese su opcion: ";
        cin >> opcion;
        switch(opcion){
            case 1:insertar();break;
            case 2:eliminar();break;
            case 3:mostrar();break;
            case 4:cout << "\nSaliendo del programa...\n";break;
            default: cout << "\nOpcion invalida. Intente nuevamente\n";
        }
    }
}

File: Colas/Arreglos/java/cola.java
import java.util.Scanner;

public class cola {
    static final int MAXSIZE = 5;
    static int[] queue = new int[MAXSIZE];
    static int front = -1, rear = -1;
    static Scanner scanner = new Scanner(System.in);

    static void insertar() {
        if (rear == MAXSIZE - 1) {
            System.out.println("\nOVERFLOW");
            return;
        }

        System.out.print("\nIngrese el elemento: ");
        int elemento = scanner.nextInt();

        if (front == -1 && rear == -1) {
            front = rear = 0;
        } else {
            rear++;
        }
        queue[rear] = elemento;
        System.out.println("\nElemento insertado correctamente.\n");
    }

    static void eliminar() {
        if (front == -1 || front > rear) {
            System.out.println("\nUnderflow");
            return;
        }

        int elemento = queue[front];
        if (front == rear) {
            front = rear = -1;
        } else {
            front++;
        }
        System.out.println("\nElemento eliminado: " + elemento + "\n");
    }

    static void mostrar() {
        if (rear == -1 || front == -1 || front > rear) {
            System.out.println("\nLa cola esta vacia.\n");
        } else {
            System.out.println("\nElementos en la cola:");
            for (int i = front; i <= rear; i++) {
                System.out.println(queue[i]);
            }
        }
    }

    public static void main(String[] args) {
        int opcion = 0;
        while (opcion != 4) {
            System.out.println("\n==========================MENU PRINCIPAL==========================");
            System.out.println("====================================================================");
            System.out.println("1. Insertar\n2. Eliminar\n3. Mostrar\n4. Salir");
            System.out.print("Ingrese su opcion: ");
            opcion = scanner.nextInt();

            switch (opcion) {
                case 1: insertar(); break;
                case 2: eliminar(); break;
                case 3: mostrar(); break;
                case 4: System.out.println("\nSaliendo del programa...\n"); break;
                default: System.out.println("\nOpcion invalida. Intente nuevamente\n"); break;
            }
        }
        scanner.close();
    }
}